tests/e2e/kubernetes/kubernetes_test.go

```diff
@@ -32,29 +32,170 @@ import (
+		{"clusterrolebindings not exist", testClusterRoleBindings(constructFoundMap(currentClusterRoleBindings), false)},
+		{"check components exist on uninstall", testComponents(false, false)},
+
+		// last parameter does not matter for the following
+		{"check mtls error", testMtls(false, false)},
+		{"status check errors out", testStatus(currentRuntimeVersion, currentDashboardVersion, false, false)},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, tc.phase)
+	}
+}
+
+func TestKubernetesUpgradeNonHA(t *testing.T) {
+	// Ensure a clean environment
+	uninstall() // does not wait for pod deletion
+	// Delete any existing CRDs Related to https://github.com/dapr/cli/issues/656
+	for _, p := range supportedUpgradePaths {
+		t.Run("delete crds "+p.previousRuntimeVersion, deleteCRD(p.previousCRDs))
+		t.Run("delete crds "+p.newRuntimeVersion, deleteCRD(p.newCRDs))
+	}
+	type testCase struct {
+		name  string
+		phase func(*testing.T)
+	}
+
+	tests := []testCase{}
+	for _, p := range supportedUpgradePaths {
+		// install previous version
+		tests = append(tests, testCase{
+			"install runtime " + p.previousRuntimeVersion,
+			testInstall(p.previousRuntimeVersion, p.previousDashboardVersion, false),
+		})
+
+		// check values
+		tests = append(tests, testCase{"crds exist " + p.previousRuntimeVersion, testCRDs(constructFoundMap(p.previousCRDs), true)})
+		tests = append(tests, testCase{"clusterroles exist " + p.previousRuntimeVersion, testClusterRoles(constructFoundMap(p.previousClusterRoles), true)})
+		tests = append(tests, testCase{"clusterrolebindings exist " + p.previousRuntimeVersion, testClusterRoleBindings(constructFoundMap(p.previousClusterRoleBindings), true)})
+
+		// initially apply a new component
+		tests = append(tests, testCase{"apply and check components exist " + p.previousRuntimeVersion, testComponents(true, true)})
+
+		tests = append(tests, testCase{"check mtls disabled " + p.previousRuntimeVersion, testMtls(true, false)})
+		tests = append(tests, testCase{
+			"status check " + p.previousRuntimeVersion,
+			testStatus(p.previousRuntimeVersion, p.previousDashboardVersion, true, false),
+		})
+
+		// upgrade to new version
+		tests = append(tests, testCase{
+			"upgrade to " + p.newRuntimeVersion,
+			testUpgrade(p.newRuntimeVersion, p.newDashboardVersion),
+		})
+
+		// check values
+		tests = append(tests, testCase{"crds exist " + p.newRuntimeVersion, testCRDs(constructFoundMap(p.newCRDs), true)})
+		tests = append(tests, testCase{"clusterroles exist " + p.newRuntimeVersion, testClusterRoles(constructFoundMap(p.newClusterRoles), true)})
+		tests = append(tests, testCase{"clusterrolebindings exist " + p.newRuntimeVersion, testClusterRoleBindings(constructFoundMap(p.newClusterRoleBindings), true)})
+
+		// on upgrade verify previously installed component exists
+		tests = append(tests, testCase{"apply and check components exist " + p.newRuntimeVersion, testComponents(true, false)})
```

Comment:
should be renamed to `check components exist on upgrade` since already the second parameter `applyComponentChanges` is false.